(feat) - O3-3197 - initial ward app

chibongho opened the pull request:
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing.en-US#contributing-guidelines) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://om.rs/o3ui).
- [ ] My work includes tests or is validated by existing tests.

## Summary
<!-- Please describe what problems your PR addresses. -->
Not much here. Just initial files for a new ward app.

## Screenshots
<!-- Required if you are making UI changes. -->

## Related Issue
<!-- Paste the link to the Jira ticket here if one exists. -->
<!-- https://issues.openmrs.org/browse/O3- -->

## Other
<!-- Anything not covered above -->

mogoodrich reviewed (approved):
LGTM, though generally defer to others...

mseaton commented on packages/esm-ward-app/src/createDashboardLink.component.tsx:
```diff
@@ -0,0 +1,35 @@
+import React, { useMemo } from 'react';
```
Remove this component.  If we need to add this to the home dashboard component in the future, and things are still not refactored out of the home app, we can add something like this in.  I'm hoping it will not be needed.

mseaton commented on packages/esm-ward-app/src/dashboard.meta.ts:
```diff
@@ -0,0 +1,6 @@
+export const wardViewDashboardMeta = {
```
Remove this file

mseaton commented on packages/esm-ward-app/src/home.comonent.tsx:
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
```
This filename is spelled incorrectly.  Additionally, I'd rather not create components that have no basis in reality yet.  We have no design for a "Home" page in the ward app.  Get rid of this component

mseaton commented on packages/esm-ward-app/src/index.ts:
```diff
@@ -0,0 +1,18 @@
+
+export const root = getSyncLifecycle(rootComponent, options);
+
+// TODO: uncomment this for the link to the ward app to appear on the left hand nav when the app is more ready
```
Remove this commented code

mseaton commented on packages/esm-ward-app/src/root.component.tsx:
```diff
@@ -0,0 +1,19 @@
+    <main>
+      <BrowserRouter basename={wardViewBasename}>
+        <Routes>
+          <Route path="/" element={<Home />} />
```
Per above, there is no current use case to have a Home component at the root of this app.  Our initial view that we are designing out will be a component that renders the patients who are currently assigned to a particular ward.  So let's create that.  Let's put that at route `/ward/${locationUuid}/patients/`, and renders a component defined as `<WardPatients location={locationUuid} />`.  Please create this WardPatients component.  You can render the location name in the top left corner as a starting point, per the mockups.

mseaton reviewed (commented):
Please ensure that the initial ward app we are committing has the following:

* Framework setup for unit tests, with an initial unit test for the initial page rendering
* Config schema (empty to start}

ibacher commented on packages/esm-ward-app/src/routes.json:
```diff
@@ -0,0 +1,29 @@
+  },
+  "extensions": [
+    {
+      "component": "root",
```
It's definitely unorthodox to have the same component as both an extension and a page.

ibacher commented on yarn.lock.orig:
This file should almost certain not be part of this PR...

chibongho commented:
Updates:
- added tests and config for the ward app.
- make the page render the location name, taken from either the location uuid in URL (if provided) or the session location.
- removed accidental added yarn.lock.orig file, stuff related to dashboard link.

![image](https://github.com/openmrs/openmrs-esm-patient-management/assets/509602/a2b19de9-6a33-4e5d-9460-d75c42f4553e)
![image](https://github.com/openmrs/openmrs-esm-patient-management/assets/509602/b0944187-0563-4fb9-98ed-d265819a78ff)

mseaton reviewed (approved):
Looks great, thanks.  I think in the ward-view, you could have used the opportunity where you have the <h1> to introduce a ward-view css file, but we can introduce that as we go.

denniskigen commented on packages/esm-ward-app/package.json:
```diff
@@ -0,0 +1,53 @@
+{
+  "name": "@openmrs/esm-ward-app",
+  "version": "7.0.0",
+  "description": "Ward and bed management module for the OpenMRS SPA",
```
```suggestion
  "description": "Ward and bed management module for O3",
```

denniskigen reviewed (approved):
Great start! Thanks
